packages/ts-moose-lib/src/consumption-helpers.ts

```diff
@@ -151,6 +152,11 @@ export class MooseClient {
       {},
     );
 
+    // We are not using the result of the ping
+    // but this ensures that if the clickhouse cloud service is idle, we
+    // wake it up.
+    this.client.ping();
```

Comment:
should we await the result, so that we know the service has woken up when we call `.query`?

Reply:
We could, but then we will add some latency every time, and it won't guarantee that the service has woken up anyway. 

So I am thinking - this makes sure the signal is sent, maybe the next step would be to do some retries on the query?

Reply:
i guess that depends on what happens when a request hit clickhouse when it's idle

Reply:
Tell me more :) how does it depend?

Reply:
So this code sends two requests one immediately after another (ping, then actual query).
This change makes sense if queries that reach clickhouse when it's "waking up" is handled properly, but the first query is not.

Reply:
Yeah - so as I understand it, the behavior is - the connection will be kept opened and help for the service to wake up and return with the response when it is woken up provided the client doesn't trigger a timeout in between.

Reply:
The difference is that I am not sure the query itself wakes the service up. I *think* you have to send a ping to wake the service up. 

I am not sure about that though. 
I also can make an option in the Boreal UI to have idle time configurable. 
Which would make this a non problem. 

Ref: https://clickhouse.com/docs/knowledgebase/clickhouse_cloud_api_usage#question